api/server/server.go

```diff
@@ -299,39 +309,28 @@ func (fha *ffsHTTPAuth) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
 	fha.cont.ServeHTTP(rw, r)
 }
 
-func startIPFSRevProxy(conf *Config, m *manager.Manager) (*http.Server, error) {
-	if conf.IPFSRevProxyAddr == "" {
-		return nil, nil
-	}
-
-	log.Info("Starting IPFS reverse proxy...")
-	ipfsIP, err := util.TCPAddrFromMultiAddr(conf.IpfsAPIAddr)
-	if err != nil {
-		return nil, fmt.Errorf("converting IPFS multiaddr to tcp addr: %s", err)
-	}
-
-	urlIPFS, err := url.Parse("http://" + ipfsIP)
-	if err != nil {
-		return nil, fmt.Errorf("generating IPFS URL for reverse proxy: %s", err)
-	}
-	rph := httputil.NewSingleHostReverseProxy(urlIPFS)
-	fha := newHTTPFFSAuthInterceptor(rph, m)
-	rp := &http.Server{
-		Addr:    conf.IPFSRevProxyAddr,
-		Handler: fha,
-	}
+func (fha *ffsHTTPAuth) IsIPFSRequest(r *http.Request) bool {
+	return len(r.Header.Get("x-ipfs-ffs-auth")) > 0
+}
 
-	go func() {
-		if err := rp.ListenAndServe(); err != nil && err != http.ErrServerClosed {
-			log.Fatalf("serving index http: %v", err)
+func createProxyServer(wrappedGRPCServer *grpcweb.WrappedGrpcServer, fha *ffsHTTPAuth, webProxyAddr string) *http.Server {
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
In here we detect if we're dealing with an IPFS request or a gRPC request and handle accordingly.